node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1449,6 +1444,14 @@ async fn fetch_finality_signatures_by_block_header(
             Some(switch_block_of_previous_era) => {
                 match switch_block_of_previous_era.next_era_validator_weights() {
                     Some(validator_weights) => {
+                        if let Some(signatures) = maybe_signatures {
+                            consensus::validate_finality_signatures(
+                                &signatures,
+                                validator_weights,
+                            )?;
```

Comment:
The commit says we would disconnect from a peer if signatures are invalid but I can't see that here.

Reply:
My oversight, good catch.
Fixed in https://github.com/casper-network/casper-node/pull/3125/commits/1622ac9ef29e794622b24d08b2437691d82e903c